source/slang/legalize-types.h

```diff
@@ -41,30 +41,34 @@ struct ImplicitDerefType;
 struct TuplePseudoType;
 struct PairPseudoType;
 struct PairInfo;
+struct LegalElementWrapping;
+struct WrappedBufferPseudoType;
 
-struct LegalType
+    /// A flavor for types or values that arise during legalization.
+enum class LegalFlavor
 {
-    enum class Flavor
-    {
-        // Nothing: a NULL type
-        none,
+        /// Nothing: an empty type or value. Equivalent to `void`.
+    none,
```

Comment:
This is a pre-existing `enum` type that I just moved around to avoid duplication. The `enum` itself dates from when I was considering a hard-line stance on the "values are `lowerCamelCase` and types are `UpperCamelCase`" convention, but it turns out in practice that `enum` cases just look better with `UpperCamelCase`.

We can do a different PR to fix up the convention, but I didn't want to have that huge diff mixed in with the functional changes here.